pkg/csi/service/wcp/controller.go

```diff
@@ -1636,23 +1638,63 @@ func (c *controller) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshot
 				return nil, logger.LogNewErrorCodef(log, codes.Internal,
 					"failed to retrieve cnsVolumeInfo for volume: %s Error: %+v", volumeID, err)
 			}
-			snapshotID, snapshotCreateTimePtr, err = common.CreateSnapshotUtil(ctx, c.manager.VolumeManager,
+			snapshotID, cnsSnapshotInfo, err = common.CreateSnapshotUtil(ctx, c.manager.VolumeManager,
 				volumeID, req.Name, &cnsvolume.CreateSnapshotExtraParams{
 					StorageClassName:           cnsVolumeInfo.Spec.StorageClassName,
 					StoragePolicyID:            cnsVolumeInfo.Spec.StoragePolicyID,
 					Namespace:                  cnsVolumeInfo.Spec.Namespace,
 					Capacity:                   cnsVolumeInfo.Spec.Capacity,
 					IsStorageQuotaM2FSSEnabled: isStorageQuotaM2FSSEnabled,
 				})
+			if err != nil {
+				return nil, logger.LogNewErrorCodef(log, codes.Internal,
+					"failed to create snapshot on volume %q: %v", volumeID, err)
+			}
+			var patch map[string]interface{}
```

Comment:
How are we patching the time info?, you need to extract that from task completion. Also, the patch of cnsvolumeinfo is not a default, it needs to be done only if the task completion timestamp is larger than current cns volumeinfo time stamp. Move things into separate functions.